### ISSUE TITLE
"Unexpected token �"

### ISSUE THREAD

**Issue by passy** [nwolverson/atom-ide-purescript#73]
I'm getting a strange error on every key press.

```
$ psc-ide-server --version
0.8.4.0
$ atom --version
Atom    : 1.7.1
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

![screenshot 2016-04-15 23 42 30](https://cloud.githubusercontent.com/assets/9906/14576745/042000e6-0364-11e6-9988-b8e37b310cd5.png)

It seems the formatting in the screenshot is more useful than plain text:

![screenshot 2016-04-15 23 45 07](https://cloud.githubusercontent.com/assets/9906/14576758/1d5b3242-0364-11e6-8e2a-d9b9f6d1d0e9.png)

My random and completely unsubstantiated suspicion is that there might be a part of the code that tries to use the default port. There's a backup program or something else occupying it and would explain the random junk showing up. At the same time, as you can see in the screenshot, on-save validation seems to be working just fine, so I guess it would be the type-ahead that's somehow buggered.

**Comment by nwolverson** [comment]
Oh.

Apparently the psc-ide port number configuration is only used to start the server, we always try to connect to 4242...

**Comment by passy** [comment]
At least my suspicion wasn't that far off then. :)
I'd like to take a stab at this but from a quick look I can't immediately see where the communication with the server happens.

**Comment by nwolverson** [comment]
It's actually via https://github.com/kRITZCREEK/purescript-psc-ide - need to pass the port. Lots of this is via https://github.com/nwolverson/purescript-ide-purescript-core

**Comment by nwolverson** [comment]
Did you start on this? I notice @kRITZCREEK has the required changes to pass a port on a branch already https://github.com/kRITZCREEK/purescript-psc-ide/commit/63638491d77ea19fdecbba09ba88dd54200c53fb

**Comment by passy** [comment]
Brilliant. No, I did not start. :)

**Comment by nwolverson** [comment]
Have a local change for this already on the rebuild-commands branch.

**Comment by nwolverson** [comment]
Fixed 6e41f996d1cab0e82e51e084bfb6597303503c54

**Comment by passy** [comment]
@nwolverson Thank you so much! 🎉
